File: Objective-C Version/AnimatedCurveDemo/AnimatedCurveDemo/LabelView/LabelView.m
//
//  LabelView.m
//  AnimatedCurveDemo
//
//  Created by yangqitao on 15/7/8.
//  Copyright (c) 2015年 Kitten Yang. All rights reserved.
//

#import "LabelView.h"
#import "UIView+Convenient.h"

#define kPullingDownString   @"下拉即可刷新..."
#define kPullingUpString     @"上拉即可刷新"
#define kReleaseString       @"松开即可刷新..."

#define kPullingString   self.state == UP ? kPullingUpString : kPullingDownString

#define LabelHeight 50

@interface LabelView()

@end

@implementation LabelView{
    UILabel *titleLabel;
}

-(id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        [self setUp];
    }
    
    return self;
}


-(void)setUp {
    self.state = DOWN;
    titleLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, self.height/2-LabelHeight/2, self.width, LabelHeight)];
    titleLabel.text = kPullingString;
    titleLabel.textColor = [UIColor blackColor];
    titleLabel.adjustsFontSizeToFitWidth = YES;
    [self addSubview:titleLabel];
}

-(void)setProgress:(CGFloat)progress{
    titleLabel.alpha = progress;
    if (!self.loading) {
        if (progress >= 1.0) {
            titleLabel.text = kReleaseString;
        }else{
            titleLabel.text = kPullingString;
        }
    }else{
        if (progress >= 0.91) {
            titleLabel.text = kReleaseString;
        }else{
            titleLabel.text = kPullingString;
        }
    }
    
}


@end

File: Swift Version/AnimatedCurveDemo-Swift/AnimatedCurveDemo-Swift/Classes/LabelView/LabelView.swift
//
//  LabelView.swift
//  AnimatedCurveDemo-Swift
//
//  Created by Kitten Yang on 1/18/16.
//  Copyright © 2016 Kitten Yang. All rights reserved.
//

import UIKit

enum PULLINGSTATE{
    case UP
    case DOWN
}

let LabelHeight: CGFloat = 50.0

class LabelView: UIView {

    let kPullingDownString = "下拉即可刷新..."
    let kPullingUpString   = "上拉即可刷新"
    let kReleaseString     = "松开即可刷新..."
    
    private var kPullingString = ""
    private var titleLabel: UILabel!
    
    var loading: Bool = false
    var progress: CGFloat = 0.0 {
        didSet{
            titleLabel.alpha = progress;
            if !loading {
                if progress >= 1.0 {
                    titleLabel.text = kReleaseString;
                }else{
                    titleLabel.text = kPullingString;
                }
            }else{
                if progress >= 0.91 {
                    titleLabel.text = kReleaseString;
                }else{
                    titleLabel.text = kPullingString;
                }
            }
        }
    }
    var state: PULLINGSTATE = .DOWN {
        didSet{
            kPullingString = (state == .UP ? kPullingUpString : kPullingDownString)
        }
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        setUp()
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    private func setUp() {
        state = .DOWN
        titleLabel = UILabel(frame: CGRectMake(0, frame.size.height/2-LabelHeight/2, frame.size.width, LabelHeight))
        titleLabel.text = kPullingString
        titleLabel.textColor = UIColor.blackColor()
        titleLabel.adjustsFontSizeToFitWidth = true
        addSubview(titleLabel)
    }
    
}
